src/main/java/com/nha/abdm/wrapper/facade/controller/HIP.java

```diff
@@ -2,33 +2,52 @@
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.nha.abdm.wrapper.hrp.CommonHelpers.ResponseHelper;
+import com.nha.abdm.wrapper.hrp.CommonHelpers.VerifyOtp;
+import com.nha.abdm.wrapper.hrp.hipInitiatedLinking.responses.AddPatient;
+import com.nha.abdm.wrapper.hrp.hipInitiatedLinking.responses.LinkRecordsResponse;
 import com.nha.abdm.wrapper.hrp.manager.WorkflowManager;
-import java.io.FileNotFoundException;
-import java.net.URISyntaxException;
-import com.nha.abdm.wrapper.hrp.serviceImpl.PatientTableService;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
+import java.io.FileNotFoundException;
+import java.net.URISyntaxException;
+import java.text.ParseException;
+import java.util.Objects;
 
 @RestController
 public class HIP {
     @Autowired
     WorkflowManager workflowManager;
     private static final Logger log = LogManager.getLogger(HIP.class);
-    @Autowired
-    PatientTableService patientTableService;
+    @PostMapping({"/link-records"})
```

Comment:
In general, I am thinking of two entities here. 1) Patient and 2) CareContext
Each entity can have its own controller like `PatientController` and `CareContextController`. In each controller we can have basic CRUD operations like `create`, `update`, `get` and `delete`. In additionally functions can be added based on the business needs like `linkCareContext`.

Reply:
Can be modified in next review